pip/req/req_set.py

```diff
@@ -570,13 +602,32 @@ def _pip_has_created_build_dir(self):
             )
         )
 
+    def _to_install(self):
+        """Create the installation order.
+
+        We install the user specified things in the order given, except when
+        dependencies require putting other things first.
```

Comment:
But again, I'd rather the doc string not imply this is really needed, or "proper".  It should be explained, as what it is, a "nice to have", not something that pip necessarily has to support.   6 months down the road, some other devs will be in here confused by this language.